### PR TITLE
Fix/matlab cli args

### PR DESCRIPTION
**Pull request by iparask**
This pull request changes the tilling matlab script to a function. It allows us to run it from the command line as

```
matlab -r 'tiling_pretraining("image","image_mask")'
```

In addition it checks if `image` and `image_mask` are strings. If not it returns an error.

### CODE REVIEW

**Review by bspitzbart — APPROVED** [review]

File: src/tiling/tiling_pretraining.m [hunk 1]
```diff
@@ -1,58 +1,70 @@
...
+    image = imread(fullfile(PathName1, FileName1));
+    
+    tmp_filename = split(FileName2,"/");
+    PathName2 = join(tmp_filename(1:end-1),"/");
+    FileName2 = tmp_filename(end);
+    river_mask = imread(fullfile(PathName2, FileName2));
+    
+    river_mask(river_mask~=0)=255;
+
+    WriteDir1 = fullfile(pwd, 'tiled multi-page image');
```
Comment (samiraucsb) [inline src/tiling/tiling_pretraining.m]: WriteDir1=fullfile(pwd, 'tiled-multi-page-image');

File: src/tiling/tiling_pretraining.m [hunk 2]
```diff
@@ -1,58 +1,70 @@
...
+    if ~exist(WriteDir1, 'dir')
+        mkdir(WriteDir1);
+    end
+
+    WriteDir2 = fullfile(pwd, 'tiled multi-page river mask');
```
Comment (samiraucsb) [inline src/tiling/tiling_pretraining.m]: WriteDir2 = fullfile(pwd, 'tiled-multi-page-river-mask');

File: src/tiling/tiling_pretraining.m [hunk 3]
```diff
@@ -1,58 +1,70 @@
...
+    if ~exist(WriteDir2, 'dir')
+        mkdir(WriteDir2);
+    end
+
+    patch_size=800;
```
Comment (samiraucsb) [inline src/tiling/tiling_pretraining.m]: This is input.

File: src/tiling/tiling_pretraining.m [hunk 4]
```diff
@@ -1,58 +1,70 @@
...
+
+    a=1:patch_size:size(image, 1);
+    b=1:patch_size:size(image, 2);
+
+    step=400;
```
Comment (samiraucsb) [inline src/tiling/tiling_pretraining.m]: This is also an input.

**Review by samiraucsb — CHANGES_REQUESTED** [review]

File: src/tiling/tiling_pretraining.m [hunk 5]
```diff
@@ -1,58 +1,65 @@
...
+    image = imread(fullfile(PathName1, FileName1));
+    
+    tmp_filename = split(FileName2,"/");
+    PathName2 = join(tmp_filename(1:end-1),"/");
+    FileName2 = tmp_filename(end);
+    river_mask = imread(fullfile(PathName2, FileName2));
+    
+    river_mask(river_mask~=0)=255;
+    tmp_folder = split(FileName1,".");
+    WriteDir1 = fullfile(pwd, 'tiled-multi-page-image', tmp_folder(1));
```
Comment (samiraucsb) [inline src/tiling/tiling_pretraining.m]: WriteDir1 = fullfile(pwd, 'tiled-multi-page-image');

File: src/tiling/tiling_pretraining.m [hunk 6]
```diff
@@ -1,58 +1,65 @@
...
+    if ~exist(WriteDir1, 'dir')
+        mkdir(WriteDir1);
+    end
+
+    WriteDir2 = fullfile(pwd, 'tiled-multi-page-river-mask', tmp_folder(1));
```
Comment (samiraucsb) [inline src/tiling/tiling_pretraining.m]: WriteDir2 = fullfile(pwd, 'tiled-multi-page-river-mask');

File: src/tiling/tiling_pretraining.m [hunk 7]
```diff
@@ -1,58 +1,65 @@
...
+        
+            clear img_window
+            clear river_mask_window
+            img_window = image(row:row+patch_size-1, col:col+patch_size-1,:);
+            river_mask_window=river_mask(row:row+patch_size-1, col:col+patch_size-1,:);
+            if nnz(sum(img_window,3))== patch_size^2
+            
+                outputFileName1 = fullfile(WriteDir1, sprintf('%.02d.tif', k));
+                saveastiff(img_window,outputFileName1);
+
```
Comment (samiraucsb) [inline src/tiling/tiling_pretraining.m:54]: movefile(fullfile(WriteDir1,'*.tif'), fullfile(WriteDir1,tmp_folder(1)));

File: src/tiling/tiling_pretraining.m [hunk 8]
```diff
@@ -1,58 +1,65 @@
...
+            if nnz(sum(img_window,3))== patch_size^2
+            
+                outputFileName1 = fullfile(WriteDir1, sprintf('%.02d.tif', k));
+                saveastiff(img_window,outputFileName1);
+            
+                river_mask_Array(:,:,1)=river_mask_window;
+                river_mask_Array(:,:,2)=imcomplement(river_mask_window);
+                outputFileName2 = fullfile(WriteDir2, sprintf('%.02d.tif', k));
+                saveastiff(river_mask_Array,outputFileName2);
+
```
Comment (samiraucsb) [inline src/tiling/tiling_pretraining.m:59]: movefile(fullfile(WriteDir2,'*.tif'), fullfile(WriteDir2,tmp_folder(1)));

File: src/training/randselect_pretraining.m [hunk 9]
```diff
@@ -9,39 +9,50 @@
...
 end
 
-FileList1 = dir(fullfile('tiling\tiled multi-page image', '*.tif'));
+
+FileList1 = dir(fullfile('../tiling/tiled-multi-page-image/*', '*.tif'))
+
 index    = randperm(numel(FileList1), floor(0.8*numel(FileList1)));
+
+for k = 1:floor(0.8*numel(FileList1))
+    movefile(fullfile(FileList1(index(k)).folder, FileList1(index(k)).name), fullfile(WriteDir1, sprintf(FileList1(index(k)).name)));
```
Comment (samiraucsb) [inline src/training/randselect_pretraining.m]:
```
movefile(fullfile(FileList1(index(k)).folder, FileList1(index(k)).name), fullfile(WriteDir1, sprintf('%.02d.tif', k)));
```

File: src/training/randselect_pretraining.m [hunk 10]
```diff
@@ -9,39 +9,50 @@
...
+end
+
+FileList2 = dir(fullfile('../tiling/tiled-multi-page-image/*', '*.tif'))
+for k = 1:numel(FileList2)
+    movefile(fullfile(FileList2(k).folder, FileList2(k).name),[WriteDir2 'Test-' FileList2(k).name]);
```
Comment (samiraucsb) [inline src/training/randselect_pretraining.m]:
```
movefile(fullfile(FileList2(k).folder, FileList2(k).name),[WriteDir2 'Test-' sprintf('%.02d.tif', k)]);
```

File: src/training/randselect_pretraining.m [hunk 11]
```diff
@@ -9,39 +9,50 @@
...
+end
+
+FileList1 = dir(fullfile('../tiling/tiled-multi-page-river-mask/*', '*.tif'));
 for k = 1:floor(0.8*numel(FileList1))
-    movefile(fullfile('tiling\tiled multi-page image', FileList1(index(k)).name), fullfile(WriteDir1, sprintf('%02d.tif', k)));
-    movefile(fullfile('tiling\tiled multi-page river mask', FileList1(index(k)).name), fullfile(WriteDir3, sprintf('%02d.tif', k)));
+    movefile(fullfile(FileList1(index(k)).folder, FileList1(index(k)).name), fullfile(WriteDir3, sprintf(FileList1(index(k)).name)));
```
Comment (samiraucsb) [inline src/training/randselect_pretraining.m]: movefile(fullfile(FileList1(index(k)).folder, FileList1(index(k)).name), fullfile(WriteDir3, sprintf('%.02d.tif', k)));

File: src/training/randselect_pretraining.m [hunk 12]
```diff
@@ -9,39 +9,50 @@
...
 end
 
-FileList2 = dir(fullfile('tiling\tiled multi-page image', '*.tif'));
+FileList2 = dir(fullfile('../tiling/tiled-multi-page-river-mask/*', '*.tif'))
 for k = 1:numel(FileList2)
-    movefile(['tiling\tiled multi-page image\' FileList2(k).name],[WriteDir2 'Test-' FileList2(k).name]);
-    movefile(['tiling\tiled multi-page river mask\' FileList2(k).name], [WriteDir4 'Test-' FileList2(k).name]);
+   movefile(fullfile(FileList2(k).folder, FileList2(k).name),[WriteDir4 'Test-' FileList2(k).name]);
```
Comment (samiraucsb) [inline src/training/randselect_pretraining.m]: movefile(fullfile(FileList2(k).folder, FileList2(k).name),[WriteDir4 'Test-' sprintf('%.02d.tif', k)]);

File: src/training/randselect_pretraining.m [hunk 13]
```diff
@@ -9,39 +9,50 @@
...
 end
 
-rmdir('tiling\tiled multi-page image','s')
-rmdir('tiling\tiled multi-page river mask','s')
+%rmdir('tiling/tiled-multi-page-image','s')
```
Comment (samiraucsb) [inline src/training/randselect_pretraining.m]: rmdir('../tiling/tiled-multi-page-image','s')

File: src/training/randselect_pretraining.m [hunk 14]
```diff
@@ -9,39 +9,50 @@
...
+%rmdir('tiling/tiled-multi-page-river-mask','s')
```
Comment (samiraucsb) [inline src/training/randselect_pretraining.m]: rmdir('../tiling/tiled-multi-page-river-mask','s')

**Review by samiraucsb — CHANGES_REQUESTED** [review]

**Comment by bspitzbart** [comment]
Who is best to make the requested changes? Samira (owns code), Giannis (owns PR), or Brad (has the most time). I am happy to do it, just checking to see if anything has been started.  This should be an easy merge.

**Comment by iparask** [comment]
That would be me, since I opened this PR.

**Review by samiraucsb — APPROVED** [review]

**Pull request merged**